Bug 19696 - Dropbox library was only returning the error code, not the description

macMikey opened the pull request:
The library is only returning half of the error message, namely the error code, but not the explanation.

montegoulding commented:
@macMikey could you add a release note and document the change to the result for synchronous commands to this PR

macMikey commented:
RN added.

montegoulding commented on extensions/script-libraries/dropbox/dropbox.livecodescript:
```diff
@@ -4318,10 +4318,11 @@ private command __API pHeader, pPost, pURL, pCallback
       if tError is empty then
          return tPostReturn for value
-      end if
-   end if
```
@macMikey I was just going to poke you about the docs patches for this but I just noticed another issue. If tsNetPost returns an error it was previously being returned but with your patch it will not be. If you reinstate the  `end if`'s above and then just make the below as follows then it will do the right thing:

```
if tPostReturn is not empty then 
   put CR&tPostReturn after tError
end if
return tError for error
```

macMikey replied:
Whoops - line out of position.  Fixed
What's missing on documentation?  It looks like my release note is in the PR.
Sorry about the mess of activity on the PR.  I tried to squash it all...

macMikey commented:
Whoops.  I thought I had.  Fail me.  I'll work on both of these, today.

macMikey committed:
Bug 19696

The library is only returning half of the error message, namely the error code, but not the explanation.

montegoulding commented:
@livecode-vulcan review ok cd23a14

livecode-vulcan commented:
💙 review by @montegoulding ok cd23a14dc4b9067a708635c7c06b1db9a6052ede

livecode-vulcan commented:
😎 test success cd23a14dc4b9067a708635c7c06b1db9a6052ede

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/1175)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/1191)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/1165)
* try-community-universal-ios-iphoneos11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos11.2/builds/118)
* try-community-universal-ios-iphonesimulator11.2: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator11.2/builds/118)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/1170)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/1192)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/1175)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/1191)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/1174)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/1184)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/1208)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/1169)
